File: dynamic/45/sol.py
from my import Solution

if __name__ == "__main__":
    solution = Solution()
    
    # Test case 1: Example from problem statement
    nums1 = [2,3,1,1,4]
    result1 = solution.jump(nums1)
    print(f"Test case 1:")
    print(f"Input: {nums1}")
    print(f"Expected: 2")
    print(f"Got: {result1}")
    assert result1 == 2, "Test case 1 failed"
    
    # Test case 2: Another example from problem statement
    nums2 = [2,3,0,1,4]
    result2 = solution.jump(nums2)
    print(f"\nTest case 2:")
    print(f"Input: {nums2}")
    print(f"Expected: 2")
    print(f"Got: {result2}")
    assert result2 == 2, "Test case 2 failed"
    
    # Test case 3: Single element array
    nums3 = [1]
    result3 = solution.jump(nums3)
    print(f"\nTest case 3:")
    print(f"Input: {nums3}")
    print(f"Expected: 0")
    print(f"Got: {result3}")
    assert result3 == 0, "Test case 3 failed"
    
    # Test case 4: Minimum jumps needed
    nums4 = [1,1,1,1]
    result4 = solution.jump(nums4)
    print(f"\nTest case 4:")
    print(f"Input: {nums4}")
    print(f"Expected: 3")
    print(f"Got: {result4}")
    assert result4 == 3, "Test case 4 failed"
    
    # Test case 5: Maximum possible jumps
    nums5 = [5,9,3,2,1,0,2,3,3,1,0,0]
    result5 = solution.jump(nums5)
    print(f"\nTest case 5:")
    print(f"Input: {nums5}")
    print(f"Expected: 3")
    print(f"Got: {result5}")
    assert result5 == 3, "Test case 5 failed"
    
    print("\nAll test cases passed!")

File: dynamic/45/sol.c
#include <stdio.h>
#include <assert.h>
#include <time.h>

// Function declaration
int jump(int* nums, int numsSize);

void runTest(int* nums, int size, int expected, int testCase) {
    printf("Test case %d:\n", testCase);
    printf("Input: [");
    for (int i = 0; i < size; i++) {
        printf("%d%s", nums[i], i < size - 1 ? "," : "");
    }
    printf("]\n");
    printf("Expected: %d\n", expected);
    
    int result = jump(nums, size);
    printf("Got: %d\n", result);
    
    assert(result == expected);
    printf("Test case %d passed!\n\n", testCase);
}

void runPerformanceTest() {
    printf("Running performance test...\n");
    
    int largeArray[10000];
    for (int i = 0; i < 10000; i++) {
        largeArray[i] = 1;
    }
    
    clock_t start = clock();
    
    for (int i = 0; i < 100; i++) {
        jump(largeArray, 10000);
    }
    
    clock_t end = clock();
    double time_spent = (double)(end - start) / CLOCKS_PER_SEC;
    printf("Performance test completed in: %f seconds\n\n", time_spent);
}

int main() {
    // Run performance test
    runPerformanceTest();
    
    // Test case 1: Example from problem statement
    int nums1[] = {2,3,1,1,4};
    runTest(nums1, 5, 2, 1);
    
    // Test case 2: Second example
    int nums2[] = {2,3,0,1,4};
    runTest(nums2, 5, 2, 2);
    
    // Test case 3: Single element array
    int nums3[] = {1};
    runTest(nums3, 1, 0, 3);
    
    // Test case 4: Minimum jumps needed
    int nums4[] = {1,1,1,1};
    runTest(nums4, 4, 3, 4);
    
    // Test case 5: Large jumps
    int nums5[] = {5,9,3,2,1,0,2,3,3,1,0,0};
    runTest(nums5, 12, 3, 5);
    
    printf("All test cases passed!\n");
    return 0;
}